File: gtm-service/scripts/send-test-email.ts
/**
 * One-shot SMTP test. Usage:
 *   npx tsx scripts/send-test-email.ts [to@example.com]
 *
 * Requires SEND_MODE=real and SMTP_USER/SMTP_PASSWORD in repo-root .env (Zoho).
 */
import { env } from "../src/config/env.js";
import { isSmtpConfigured, sendEmail } from "../src/mail/smtp.js";

async function main() {
  const to =
    process.argv[2]?.trim() ||
    env.testToEmail ||
    "faruquitaymur@gmail.com";

  console.log("SMTP test");
  console.log(`  SEND_MODE=${env.sendMode}`);
  console.log(`  SMTP_HOST=${env.smtpHost}:${env.smtpPort}`);
  console.log(`  SMTP_USER=${env.smtpUser}`);
  console.log(`  SMTP_FROM=${env.senderEmail}`);
  console.log(`  configured=${isSmtpConfigured()}`);
  console.log(`  to=${to}`);

  if (env.sendMode !== "real") {
    console.error("FAIL: Set SEND_MODE=real in .env to actually send.");
    process.exit(1);
  }
  if (!isSmtpConfigured()) {
    console.error(
      "FAIL: Set SMTP_USER and SMTP_PASSWORD (Zoho) in the repo-root .env.",
    );
    process.exit(1);
  }

  const result = await sendEmail({
    to,
    subject: "SMTP test – Speaker Signal GTM",
    text: "This is a test email from the Speaker Signal GTM backend. SMTP is working.",
    html:
      "<p>This is a test email from the <strong>Speaker Signal</strong> GTM backend.</p>" +
      "<p>SMTP is working.</p>",
  });
  console.log(`OK: sent (${result.mode}) → ${to}`);
}

main().catch((err) => {
  console.error("FAIL:", err instanceof Error ? err.message : err);
  process.exit(1);
});

File: agents/agent3/send_test_email.py
"""Send one SMTP test email via Agent 3 mailer.

Usage (from repo root):
  python -m agents.agent3.send_test_email
  python -m agents.agent3.send_test_email you@example.com

Requires SEND_MODE=real and SMTP_PASSWORD (Gmail App Password) in .env.
"""
from __future__ import annotations

import os
import sys

from . import config, mailer


def main() -> int:
    to = (
        (sys.argv[1] if len(sys.argv) > 1 else "").strip()
        or os.getenv("TEST_TO_EMAIL", "").strip()
        or "kirill.cheldishkin2105@gmail.com"
    )
    print("SMTP test (agent3)")
    print(f"  SEND_MODE={config.SEND_MODE}")
    print(f"  SMTP_HOST={config.SMTP_HOST}:{config.SMTP_PORT}")
    print(f"  SMTP_USER={config.SMTP_USER}")
    print(f"  password_set={bool(config.SMTP_PASSWORD)}")
    print(f"  to={to}")

    if config.SEND_MODE != "real":
        print("FAIL: Set SEND_MODE=real in .env to actually send.", file=sys.stderr)
        return 1
    if not config.SMTP_PASSWORD:
        print(
            "FAIL: Set SMTP_PASSWORD to a Gmail App Password "
            "(https://myaccount.google.com/apppasswords).",
            file=sys.stderr,
        )
        return 1

    mode = mailer.deliver(
        to,
        "SMTP test – Speaker Signal Agent 3",
        "This is a test email from Agent 3. SMTP is working.",
    )
    print(f"OK: sent ({mode}) → {to}")
    return 0


if __name__ == "__main__":
    raise SystemExit(main())
